src/mesh/BoundingBox.cpp

```diff
@@ -44,7 +76,7 @@ bool BoundingBox::operator==(const BoundingBox &otherBB) const
 bool BoundingBox::empty() const
 {
   for (int i = 0; i < _dimensions; ++i) {
-    if (_bounds[2 * i] > _bounds[2 * i + 1]) {
+    if (_boundMin[i] > _boundMax[i]) {
```

Comment:
To refer to your question: 

> but here comes a question: can 3D BoundingBox has zero length along one of the axes, i.e. has zero volume

I guess having 'dead' directions should still be a valid bounding box. We have such cases (also 2D dimensions with only a 1D interface, i.e., the bounding box would be 1D). In my opinion we have two options here: either we clearly return `false` in case we have at least one vertex inside the box (min and max bounds are the same) or we introduce a slightly different version of this function like `getActiveDimensions()`, which returns an `int` indicating the non-zero dimensions. The `empty` function is anyway mainly used for asserts, so that we can then ask in the code `PRECICE_ASSERT(getActiveDimensions >0)` or similar.

Reply:
Thank @davidscn for your response : )

Quick questions please: what are 'dead' directions? What are them for? Does they relate to `deadAxis` as an input arguments in the function `getArea` of BoundingBox?
```cpp
double BoundingBox::getArea(std::vector<bool> deadAxis);
```

Reply:
No, I was just referring to directions which have zero expansion, i.e., min == max

Reply:
We tried to implement the getActiveDimensions, but we think the function would be the same as the getDimension method, because, in the new BB, the non-zero dimensions are the number of dimensions instantiated in the Eigen::VectorXd. @davidscn

Reply:
We choose to implement empty() directly via Eigen's ```isZero()``` on difference of two vertices (shown below) instead of implementing getActiveDimensions() since it's quite simple and short.
```cpp
/// empty() returning true means that the length in every BoundingBox's dimension is equal to zero
bool BoundingBox::empty() const
{
  return (_boundMax - _boundMin).isZero();
}
```

@davidscn An additional question as regards empty() please. Which of member functions should be added the empty() check, i.e. ``` PRECICE_ASSERT(!empty(), "The BoundingBox is empty, i.e. it has zero area or zero volume.");```?

Reply:
What happens here if we have single axis being in the default state? Then we would have negative entries in the result and `isZero` would return `false`, which is not what we are looking for, right?

>  PRECICE_ASSERT(!empty(), "The BoundingBox is empty, i.e. it has zero area or zero volume.")

You can add it whenever you think it makes sense, will have a look in the final review.

Reply:
Done.